draft-ietf-masque-h3-datagram.md

```diff
@@ -31,66 +31,74 @@ author:
 
 --- abstract
 
-The QUIC DATAGRAM extension provides application protocols running over QUIC
-with a mechanism to send unreliable data while leveraging the security and
-congestion-control properties of QUIC. However, QUIC DATAGRAM frames do not
-provide a means to demultiplex application contexts. This document describes how
-to use QUIC DATAGRAM frames with HTTP/3 by association with HTTP requests.
-Additionally, this document defines the Capsule Protocol that can convey
-datagrams over prior versions of HTTP.
+This document describes HTTP Datagrams, a convention for conveying multiplexed,
+potentially unreliable datagrams inside an HTTP connection.
+
+In HTTP/3, HTTP Datagrams can be conveyed natively using the QUIC DATAGRAM
+extension. When the QUIC DATAGRAM frame is unavailable or undesirable, they can
+be sent using the Capsule Protocol, a more general convention for conveying data
+in HTTP connections.
+
+Both are intended for use by HTTP extensions, not applications.
 
 
 --- middle
 
 # Introduction {#intro}
 
-The QUIC DATAGRAM extension {{!DGRAM=I-D.ietf-quic-datagram}} provides
-application protocols running over QUIC {{!QUIC=RFC9000}} with a mechanism to
-send unreliable data while leveraging the security and congestion-control
-properties of QUIC. However, QUIC DATAGRAM frames do not provide a means to
-demultiplex application contexts. This document describes how to use QUIC
-DATAGRAM frames with HTTP/3 {{!H3=I-D.ietf-quic-http}} by association with HTTP
-requests. Additionally, this document defines the Capsule Protocol that can
-convey datagrams over prior versions of HTTP.
-
+HTTP extensions sometimes need to access underlying transport protocol features
+such as unreliable delivery (as offered by {{!DGRAM=I-D.ietf-quic-datagram}})
+to enable desirable features like an unreliable version of the CONNECT method,
+and unreliable delivery in WebSockets {{?RFC6455}} (or its successors).
 
-This document is structured as follows:
+In {{datagrams}}, this document describes HTTP Datagrams, a convention that
+associates non-content data with HTTP requests in order to support the
+bidirectional and possibly multiplexed exchange of datagrams inside an HTTP
+connection. They are intended for use by HTTP extensions (such as the CONNECT
```

Comment:
'non-content' raises a lot of questions. It might be better to be more direct, e.g.,

```suggestion
In {{datagrams}}, this document describes HTTP Datagrams, a convention that
supports the bidirectional and possibly multiplexed exchange of data inside an HTTP
connection. While HTTP datagrams are associated with HTTP requests, they are not 
part of message content; instead, they are intended for use by HTTP extensions (such as the CONNECT
```

Reply:
Fixed